ui/src/app/app.tsx

```diff
@@ -101,7 +102,19 @@ requests.onError.subscribe(async err => {
         // If basehref is the default `/` it will become an empty string.
         const basehref = document.querySelector('head > base').getAttribute('href').replace(/\/$/, '');
         if (isSSO) {
-            window.location.href = `${basehref}/auth/login?return_url=${encodeURIComponent(location.href)}`;
+            const authSettings = await services.authService.settings();
+            const ctx = React.useContext(Context);
```

Comment:
It is better to move this into the if that handles the error handling, cause needed only in case of error.

Reply:
This section had to be moved to a location that I could access the react component state. It should also clean this up.